basic-testing/index.html

```diff
@@ -3,170 +3,158 @@
+  const jsPsych = initJsPsych({
+    on_finish: function () {
+      test();
+    },
+  });
 
-const jsPsych = initJsPsych({
-  on_finish: function(){
-    test();
-  }
-});
-
-const instructions = {
-  type: jsPsychHtmlButtonResponse,
-  stimulus: `
+  const instructions = {
+    type: jsPsychHtmlButtonResponse,
+    stimulus: `
     <p>This experiment measures how fast you can identify whether a number is odd or even.</p>
     <p>You'll see a number appear on the screen. As quickly as you can:</p>
     <p>Press the A key if the number is odd.</p>
     <p>Press the L key if the number is even.</p>
   `,
-  choices: ['I understand the instructions']
-}
-
-const instruction_check = {
-  timeline: [
-    {
-      type: jsPsychHtmlButtonResponse,
-      stimulus: ()=>{ return `
-        <p class="stimulus">${jsPsych.timelineVariable('number')}</p>
-      `},
-      choices: ['A', 'L'],
-      prompt: '<p>To see if you understood the instructions, which key should you press if you see this number?</p>',
-      data: {
-        task: 'instructions-check'
+    choices: ["I understand the instructions"],
+  };
+
+  const instruction_check = {
+    timeline: [
+      {
+        type: jsPsychHtmlButtonResponse,
+        stimulus: () => {
+          return `
+        <p class="stimulus">${jsPsych.timelineVariable("number")}</p>
+      `;
+        },
+        choices: ["A", "L"],
+        prompt:
+          "<p>To see if you understood the instructions, which key should you press if you see this number?</p>",
+        data: {
+          task: "instructions-check",
+        },
+        on_finish: (data) => {
+          data.correct = data.response === jsPsych.timelineVariable("answer");
+        },
       },
-      on_finish: (data) => {
-        data.correct = data.response == jsPsych.timelineVariable('answer');
-      }
-    }
-  ],
-  timeline_variables: [
-    {number: 3, answer: 0},
-    {number: 6, answer: 1}
-  ],
-  randomize_order: true  
-}
-
-const instructions_check_fail = {
-  timeline: [
-    {
-      type: jsPsychHtmlButtonResponse,
-      stimulus: `<p>You did not answer the questions correctly.</p>
+    ],
+    timeline_variables: [
+      { number: 3, answer: 0 },
+      { number: 6, answer: 1 },
+    ],
+    randomize_order: true,
+  };
+
+  const instructions_check_fail = {
+    timeline: [
+      {
+        type: jsPsychHtmlButtonResponse,
+        stimulus: `<p>You did not answer the questions correctly.</p>
         <p>We'll show you the instructions again.</p>
       `,
-      choices: ['Got it']
-    }
-  ],
-  conditional_function: ()=>{
-    const ic_data = jsPsych.data.get().filter({task: 'instructions-check'}).last(2);
-    return !ic_data.select('correct').all((value)=>{return value==true});
-  }
-}
+        choices: ["Got it"],
+      },
+    ],
+    conditional_function: () => {
+      const ic_data = jsPsych.data.get().filter({ task: "instructions-check" }).last(2);
+      return !ic_data.select("correct").all(Boolean);
```

Comment:
I don't understand why this works 🤔

Reply:
The `Boolean` constructor is nothing but a function that, given any value, returns `true` if it is "truthy" (`== true`) and `false` otherwise :stuck_out_tongue_winking_eye: